buildSrc/src/main/java/org/elasticsearch/gradle/util/Util.java

```diff
@@ -129,15 +129,39 @@ public static FileTree getJavaTestAndMainSourceResources(Project project, Action
     }
 
     /**
+     * Returns the source set for either the YamlTests or the standard test source set. Preference goes to YamlTest source set if it exists.
+     *
      * @param project The project to look for test Java resources.
      * @return An Optional that contains the Java test SourceSet if it exists.
      */
+    public static Optional<SourceSet> getTestSourceSet(Project project) {
```

Comment:
As mentioned in my comment, I don't think we need to support this case. We should _require_ that all YAML tests live in the appropriate source set. This awkward conditional stuff is a great example of why.

Also, what happens if a project has both tests types? Which source set should this method return? As part of splitting these types of tests up there is no longer a canonical "test source set", as any given project might (and likely will) have several.

Reply:
Discussed in person. Since this is used primarily for copy around of REST resources, I will remove this method in favor of having the REST resources be explicit about which source set they will use dependent on the application (or not) of the new YAML test task. 

Also, this PR updated to use this with test conventions task, but that isn't needed for custom test source sets.

There will still be a todo to circle back around and remove the conditional aspect of which test source set is used for copying around REST resources, but that can only be done after all existing usages of the java test source set have been addressed. This includes ensuring that the clients teams won't break on moving tests around, as well all internal usages where YAML tests can live in the java test source set. While plugin developers may be non-passively impacted by requiring the new YAML plugin, the migration path is pretty simple and would only impact those that leverage YAML testing (suspected to be very few).

Reply:
done ... will add summary to main comment.